src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,145 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private Object[] elementArray;
+    private int size;
+
+    public ArrayList() {
+        this.elementArray = new Object[DEFAULT_CAPACITY];
+    }
+
+    private Object[] grow() {
+        int newLength = (size * 3) / 2 + 1;
+        Object[] newArray = new Object[newLength];
+        System.arraycopy(elementArray, 0, newArray, 0, size);
+        return elementArray = newArray;
+    }
+
+    private Object[] cutBefore(int index) {
+        Object[] cutArray = new Object[index];
+        System.arraycopy(elementArray, 0, cutArray, 0, index);
+        return cutArray;
+    }
+
+    private Object[] cutAfter(int index) {
+        Object[] cutArray = new Object[size - index];
+        System.arraycopy(elementArray, index, cutArray, 0, size - index);
+        return cutArray;
+    }
+
     @Override
     public void add(T value) {
-
+        if (elementArray.length == size) {
+            grow();
+        }
+        elementArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (elementArray.length == size) {
+            grow();
+        }
+        if (index == size) {
+            add(value);
+        } else if (index < size && index >= 0) {
+            Object[] beforeArray = cutBefore(index);
+            Object[] afterArray = cutAfter(index);
+            Object[] newArray = new Object[elementArray.length];
+            System.arraycopy(beforeArray, 0, newArray, 0, beforeArray.length);
+            newArray[index] = value;
+            System.arraycopy(afterArray, 0, newArray, index + 1, afterArray.length);
+            elementArray = newArray;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("index exception");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
```

Comment:
Lets use here for each with method add(value)

Reply:
I do not see for each...